session.go

```diff
@@ -4,43 +4,40 @@ import (
 	"reflect"
 	"sync"
 
-	"upper.io/db"
+	"upper.io/db.v2"
+	"upper.io/db.v2/postgresql" // TODO: Figure out how to remove this and make it generic.
 )
 
 type Session interface {
-	db.Tx
+	postgresql.Tx
 	Store(interface{}) Store
 	Find(...interface{}) db.Result
 	Save(Model) error
 	Delete(Model) error
-	NewTransaction() (Session, error)
+	Tx() (Session, error)
 	ContinueTransaction() (Session, error, bool)
 }
 
 type session struct {
-	db.Database
+	postgresql.Database
 	stores     map[string]*store
 	storesLock sync.Mutex
 }
 
+// Open connects to a database.
 func Open(adapter string, url db.ConnectionURL) (Session, error) {
-	conn, err := db.Open(adapter, url)
+	conn, err := postgresql.Open(url)
 	if err != nil {
 		return nil, err
 	}
 
-	sess := &session{
-		Database: conn,
-		stores:   make(map[string]*store),
-	}
-
-	return sess, nil
+	return &session{Database: conn, stores: make(map[string]*store)}, nil
 }
 
-// NewTransaction creates and returns a session that runs within a transaction
```

Comment:
We have a `NewTransaction` method on v2 and this is named the same, so it creates a conflict. I renamed this into `Tx()`. Stores already have a `Tx(tx)` so that might not be that difficult to remember.